External/Plugins/ASCompletion/PluginMain.cs

```diff
@@ -446,104 +446,108 @@ public void HandleEvent(Object sender, NotifyEvent e, HandlingPriority priority)
-                                string iterator = ASComplete.FindFreeIterator(ASContext.Context, ASContext.Context.CurrentClass, result.Context);
-                                details.Add("ItmUniqueVar", iterator);
+                                // resolve current element
+                                Hashtable details = ASComplete.ResolveElement(sci, null);
                                 te.Value = ArgumentsProcessor.Process(te.Value, details);
-                            }
-                        }
-                        break;
 
-                    // menu commands
-                    case EventType.Command:
-                        string command = (e as DataEvent).Action ?? "";
-                        if (command.StartsWith("ASCompletion."))
-                        {
-                            string cmdData = (e as DataEvent).Data as string;
-                            // run MTASC
-                            if (command == "ASCompletion.CustomBuild")
-                            {
-                                if (cmdData != null) ASContext.Context.RunCMD(cmdData);
-                                else ASContext.Context.RunCMD("");
-                                e.Handled = true;
+                                if (te.Value.IndexOf("$") >= 0 && reCostlyArgs.IsMatch(te.Value))
+                                {
+                                    ASResult result = ASComplete.CurrentResolvedContext.Result ?? new ASResult();
+                                    details = new Hashtable();
+                                    // Get closest list (Array or Vector)
+                                    string closestListName = "", closestListItemType = "";
+                                    ASComplete.FindClosestList(ASContext.Context, result.Context, sci.CurrentLine, ref closestListName, ref closestListItemType);
+                                    details.Add("TypClosestListName", closestListName);
+                                    details.Add("TypClosestListItemType", closestListItemType);
+                                    // get free iterator index
+                                    string iterator = ASComplete.FindFreeIterator(ASContext.Context, ASContext.Context.CurrentClass, result.Context);
+                                    details.Add("ItmUniqueVar", iterator);
+                                    te.Value = ArgumentsProcessor.Process(te.Value, details);
+                                }
                             }
+                            break;
 
-                            // build the SWF using MTASC
-                            else if (command == "ASCompletion.QuickBuild")
+                        // menu commands
+                        case EventType.Command:
+                            de = e as DataEvent;
+                            string command = de.Action ?? "";
+                            if (command.StartsWith("ASCompletion.", StringComparison.Ordinal))
```

Comment:
Can I make a string extension class in `PluginCore.Utilities` with `StartsWithOrdinal`, `EndsWithOrdinal`, `IndexOfOrdinal` and `LastIndexOfOrdinal`? It would make the code much more readable.